recipe-appTests/listRecipesTest.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  listRestaurantsTest.swift
+//  recipe-appTests
+//
+//  Created by Arthur Hermann on 27/11/2024.
+//
+
+import XCTest
+import Testing
+@testable import recipe_app
+
+final class listRestaurantsTest: XCTestCase {
+    
+    func testRecipeListNotEmpty() {
+        DispatchQueue.global(qos: .background).async {
+            let vm = RecipeListViewModel()
+            vm.searchRecipe("pasta")
+            XCTAssertFalse(vm.recipes.isEmpty)
```

Comment:
The `RecipeListViewModel` keeps its state in the `recipes` property, so if multiple tests run at the same time, they might mess with each other. This can cause weird issues like race conditions. One of the reasons why your tests were passing even tho your fetch request wasn't even going through.

Try isolating the view model for each test or mocking its dependencies to keep tests independent.